include/userobjects/ParticleStepperBase.h

```diff
@@ -0,0 +1,78 @@
+//* This file is part of FENIX: Fusion Energy Integrated Multiphys-X,
+//* A multiphysics application for modeling plasma facing components
+//* https://github.com/idaholab/fenix
+//*
+//* FENIX is powered by the MOOSE Framework
+//* https://www.mooseframework.inl.gov
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE Includes
+#include "Ray.h"
+#include "GeneralUserObject.h"
+#include "VariableSampler.h"
+#include "RayTracingStudy.h"
+
+class ParticleStepperBase : public GeneralUserObject
+{
+public:
+  ParticleStepperBase(const InputParameters & parameters);
+
+  static InputParameters validParams();
+  /**
+   * Unused methods
+   */
+  ///@{
+  virtual void initialize() override {}
+  virtual void finalize() override {}
+  virtual void execute() override {}
```

Comment:
```suggestion
  virtual void initialize() override final {}
  virtual void finalize() override final {}
  virtual void execute() override final {}
```

With this, you'll get an error if you try to override in a derived class. Just adds more intent